src/devices/bus/centronics/etibuffer.cpp

```diff
@@ -0,0 +1,369 @@
+	map(0x0, 0xffff).r(FUNC(etiprintbuffer_device::eti_status_r));
+}
+
+//--------------------------------------------------
+//  memory and io functions
+//--------------------------------------------------
+
+uint8_t etiprintbuffer_device::eti_status_r(offs_t offset)
+{
+	return ((m_strobereceived ? 1 : 0 ) << 6) | // d6 is m_strobe status
+			((m_busy          ? 0 : 1 ) << 7);  // d7 is not busy output
+}
+
+uint8_t etiprintbuffer_device::eti_read_1000(offs_t offset)
+{
+	output_ack(0);
+	m_strobereceived = false;
+	m_bufferready_led = !m_strobereceived;
+
+	output_busy(m_strobereceived);
+	m_ack_timer->adjust(attotime::from_usec(ACK_DELAY), TIMER_ACK);
+	return m_datalatch;
+}
+
+void etiprintbuffer_device::eti_write_2000(offs_t offset, uint8_t data)
+{
+	m_ctx_data_out->write(data);
+}
+
+void etiprintbuffer_device::eti_write_3000(offs_t offset, uint8_t data)
+{
+	m_ctx->write_strobe(CLEAR_LINE);
+	m_strobeout_timer->adjust(attotime::from_usec(STROBE_DELAY), TIMER_STROBEOUT);
+}
+
+//--------------------------------------------------
+//  device init
+//--------------------------------------------------
+
+
+void etiprintbuffer_device::device_start()
+{
+	m_printerready_led.resolve();
+	m_bufferready_led.resolve();
+	m_ack_timer = timer_alloc(TIMER_ACK);
+	m_strobeout_timer = timer_alloc(TIMER_STROBEOUT);
+
+	save_item(NAME(m_ram));
+	save_item(NAME(m_data));
+	save_item(NAME(m_datalatch));
+	save_item(NAME(m_strobereceived));
+	save_item(NAME(m_strobe));
+	save_item(NAME(m_busy));
+}
+
+void etiprintbuffer_device::device_reset()
+{
+	address_space &mem = m_maincpu->space(AS_PROGRAM);
+	u16 ram16kblocks = ioport("CONFIG")->read() >> 2;
+	if (ram16kblocks) mem.install_ram(0x4000,0x4000+ram16kblocks*0x4000-1,m_ram);
+
+	output_busy(0);
+	output_fault(1);
+	output_ack(1);
+	output_select(1);
+	output_perror(0);
+	m_ctx->write_strobe(ASSERT_LINE);
+}
+
+//--------------------------------------------------
+//  device mconfig
+//--------------------------------------------------
+
+void etiprintbuffer_device::device_add_mconfig(machine_config &config)
+{
+	Z80(config, m_maincpu, 2_MHz_XTAL);
+	m_maincpu->set_addrmap(AS_PROGRAM, &etiprintbuffer_device::etiprintbuffer_device_memmap);
+	m_maincpu->set_addrmap(AS_IO,      &etiprintbuffer_device::etiprintbuffer_device_iomap);
+
+	CENTRONICS(config, m_ctx, centronics_devices, "printer");
+	m_ctx->busy_handler().set(FUNC(etiprintbuffer_device::busy_w));
+
+	OUTPUT_LATCH(config, m_ctx_data_out);
+	m_ctx->set_output_latch(*m_ctx_data_out);
+
+#ifdef DEBUG_ETI_BUFFER
+	// video hardware
+	screen_device &screen(SCREEN(config, m_screen, SCREEN_TYPE_RASTER));
+	screen.set_refresh_hz(60);
+	screen.set_vblank_time(ATTOSECONDS_IN_USEC(0));
+	screen.set_size(SCREEN_WIDTH, SCREEN_HEIGHT);
+	screen.set_visarea(0, SCREEN_WIDTH-1, 0, SCREEN_HEIGHT-1);
+	screen.set_screen_update(FUNC(etiprintbuffer_device::screen_update_etibuffer));
+	screen.set_physical_aspect(1,40); // long and slender
+#endif
```

Comment:
If you want a graph display, use a layout with embedded Lua script.  No fake screens.

Reply:
ok, I'll just take that out for now